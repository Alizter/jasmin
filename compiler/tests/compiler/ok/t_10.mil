extern fn test(reg x_in : u64) -> reg u64 {
  reg x, y, z1, z2, x_out : u64;
  flag cf : bool;

  z1 = 5;
  x = x_in;
  y = 10;
  do {
    x += z1;
    y += 1;
  } while !cf;
  
  if cf {
    x = z1;
  } else {
    x = z2;
  }

  x += y;
  x += x;
  x_out = x;
  return x_out;
}
