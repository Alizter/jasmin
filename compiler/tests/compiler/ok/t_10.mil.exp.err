Parsing tests/compiler/ok/t_10.mil as MIL

Processed file tests/compiler/ok/t_10.mil
