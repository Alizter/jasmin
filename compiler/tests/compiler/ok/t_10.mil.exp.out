>> renum:
param n : u64;

extern fn test() {
  i : inline u64
  x : stack u64[(n : u64)]
  (x : stack u64[(n : u64)])[0] = 0;
  for (i : inline u64) in 0..$(n : u64) {
    if ($(n : u64) = 10) {
      (x : stack u64[(n : u64)])[$(n : u64) - 1] += 1;
    } else {
      (x : stack u64[(n : u64)])[$(n : u64) - 1] += 1;
    }
  }
}


total transformation time: REMOVED_TIME

total time (with save/print): REMOVED_TIME
